plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevControllerTest.kt

```diff
@@ -46,43 +74,72 @@ class FeatureDevControllerTest : FeatureDevTestBase() {
     private lateinit var spySession: Session
     private lateinit var featureDevClient: FeatureDevClient
 
-    private val tabId = "tabId"
+    private val newFileContents = listOf(
+        NewFileZipInfo("test.ts", "This is a comment", false),
+        NewFileZipInfo("test2.ts", "This is a rejected file", true)
+    )
+    private val deletedFiles = listOf(
+        DeletedFileInfo("delete.ts", false),
+        DeletedFileInfo("delete2.ts", true)
+    )
+    private val references = listOf(CodeReference(information = "test"))
 
     @Before
     override fun setup() {
         super.setup()
         featureDevClient = mock()
         messenger = mock()
         chatSessionStorage = mock()
+        projectRule.project.replaceService(FeatureDevClient::class.java, featureDevClient, disposableRule.disposable)
         appContext = mock<AmazonQAppInitContext> {
-            on { project }.thenReturn(project)
+            on { project }.thenReturn(projectRule.project)
             on { messagesFromAppToUi }.thenReturn(messenger)
         }
-        authController = spy(AuthController())
+        authController = mock()
+        whenever(authController.getAuthNeededStates(any())).thenReturn(AuthNeededStates(amazonQ = null))
+        spySession = spy(Session(testTabId, projectRule.project))
+
+        mockkStatic(
+            MessagePublisher::sendAnswer,
+            MessagePublisher::sendSystemPrompt,
+            MessagePublisher::sendUpdatePlaceholder,
+            MessagePublisher::sendChatInputEnabledMessage,
+            MessagePublisher::sendCodeResult,
+            MessagePublisher::updateFileComponent
+        )
+
         mockkStatic("software.aws.toolkits.jetbrains.services.amazonqFeatureDev.util.UploadArtifactKt")
         every { uploadArtifactToS3(any(), any(), any(), any(), any()) } just runs
 
         controller = FeatureDevController(appContext, chatSessionStorage, authController)
     }
 
+    @After
+    fun clear() {
+        unmockkAll()
+    }
+
     @Test
     fun `test new tab opened`() {
-        val message = IncomingFeatureDevMessage.NewTabCreated("new-tab-created", tabId)
+        authController = spy(AuthController())
+        controller = FeatureDevController(appContext, chatSessionStorage, authController)
+
+        val message = IncomingFeatureDevMessage.NewTabCreated("new-tab-created", testTabId)
         spySession = spy(Session("tabId", project))
         whenever(chatSessionStorage.getSession(any(), any())).thenReturn(spySession)
 
         runTest {
             controller.processNewTabCreatedMessage(message)
         }
-        verify(authController, times(1)).getAuthNeededStates(project)
-        verify(chatSessionStorage, times(1)).getSession(tabId, project)
+        mockitoVerify(authController, times(1)).getAuthNeededStates(project)
```

Comment:
This need not be changed; it serves the same purpose.

Reply:
I needed to rename the reference as the kotlin testing library has the same name for the verify function.